stream/src/main/scala/org/apache/pekko/stream/impl/fusing/GraphStages.scala

```diff
@@ -381,27 +381,25 @@ import pekko.stream.stage._
     override def initialAttributes: Attributes = DefaultAttributes.futureSource
     override def createLogic(attr: Attributes) =
       new GraphStageLogic(shape) with OutHandler {
-        def onPull(): Unit = {
+        override def preStart(): Unit = {
           future.value match {
             case Some(completed) =>
               // optimization if the future is already completed
-              onFutureCompleted(completed)
+              handle(completed)
```

Comment:
This will can break up some code, because we change the existing behavior. 
@mdedetrich 

The `FailedSource` failing the stream ASAP. but maybe not this one?

Especially when combine it with `flatmapConcat` with parallism materialization.

Reply:
@nvollmar @pjfanning What's your option here